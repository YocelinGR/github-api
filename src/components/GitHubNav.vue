<template>
    <header role="banner">
      <div class="container-fluid row">
        <div class="col-md-1">
          <i class='fab fa-github icon-nav' style='font-size:40px;color:white'></i>
        </div>
        <div class="col-md-9">
          <nav class="navbar navbar-expand-lg git-nav">
            <div id="navbarSupportedContent" class="collapse navbar-collapse">
              <form class="form-inline my-2 my-lg-0">
                <input class="form-control mr-sm-2" type="search" placeholder="Search or jump to..." aria-label="Search">
                <button class="btn btn-outline-success my-2 my-sm-0" type="submit"><i class='fas fa-slash' style='font-size:18px;color:green'></i></button>
              </form>
              <ul class="navbar-nav mr-auto">
                <li class="nav-item active">
                  <a class="nav-link" href="#">Pull Request <span class="sr-only">(current)</span></a>
                </li>
                <li class="nav-item">
                  <a class="nav-link" href="#">Issues</a>
                </li>
                <li class="nav-item">
                  <a class="nav-link" href="#">Marketplace</a>
                </li>
                <li class="nav-item">
                  <a class="nav-link" href="#">Explore</a>
                </li>
              </ul>
            </div>
          </nav>
        </div>
        <div class="col-md-1">
          <img :src="user.avatarUrl" alt="user photo">
        </div>
      </div>
    </header>
</template>

<script>
import gql from 'graphql-tag';

const getUser = gql`
  query user {
  user(login: "YocelinGR"){
    avatarUrl(size: 30),
    id
  }
}
`;
export default {
  data: () => ({
    user: {},
  }),
  apollo: {
    user: {
       query: getUser,
      }
    }
 }
</script>

<style scoped>
  header {
    position: relative;
    padding-top: 12px;
    padding-bottom: 12px;
    font-family: Helvetica,Arial,sans-serif;
    font-size: 14px !important;
    color: rgb(206, 209, 212);
    background-color: #24292e;
  }
  .git-nav, .icon-nav {
    position: relative;
    box-sizing: border-box;
    display: flex;
    flex: 1 1 auto;
    justify-content: space-between;
    color: white;
    background-color: #24292e;
  }
  a, i {
    color: white !important;
  }
  img {
    float: right;
    padding-top: 12px;
    padding-bottom: 12px;
  }
</style>
