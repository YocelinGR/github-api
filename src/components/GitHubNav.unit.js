import GitHubNav from './GitHubNav.vue'
import { createLocalVue } from '@vue/test-utils'

describe('@components/GitHubNav', () => {
  const localVue = createLocalVue()
  it('GitHubNav is a vue instance', () => {
    const wrapper = shallowMount(GitHubNav, { localVue })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('Displays user profile', () => {
    const wrapper = shallowMount(GitHubNav, { localVue })
    wrapper.setData({
      user: {
        avatarUrl: "https://avatars3.githubusercontent.com/u/39833045?s=30&v=4",
        bio: "Front End Developer egresada de Laboratoria e\r\nIngeniera en Comunicaciones y Electrónica.\r\nApasionada por la programación y fiel creyente de\r\nla mejora continua",
        id: "MDQ6VXNlcjM5ODMzMDQ1"
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
