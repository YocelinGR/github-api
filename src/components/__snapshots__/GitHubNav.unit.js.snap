// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/GitHubNav Displays user profile 1`] = `
<header
  role="banner"
>
  <div
    class="container-fluid row"
  >
    <div
      class="col-md-1"
    >
      <i
        class="fab fa-github icon-nav"
        style="font-size: 40px; color: white;"
      />
    </div>
     
    <div
      class="col-md-9"
    >
      <nav
        class="navbar navbar-expand-lg git-nav"
      >
        <div
          class="collapse navbar-collapse"
          id="navbarSupportedContent"
        >
          <form
            class="form-inline my-2 my-lg-0"
          >
            <input
              aria-label="Search"
              class="form-control mr-sm-2"
              placeholder="Search or jump to..."
              type="search"
            />
             
            <button
              class="btn btn-outline-success my-2 my-sm-0"
              type="submit"
            >
              <i
                class="fas fa-slash"
                style="font-size: 18px; color: green;"
              />
            </button>
          </form>
           
          <ul
            class="navbar-nav mr-auto"
          >
            <li
              class="nav-item active"
            >
              <a
                class="nav-link"
                href="#"
              >
                Pull Request 
                <span
                  class="sr-only"
                >
                  (current)
                </span>
              </a>
            </li>
             
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="#"
              >
                Issues
              </a>
            </li>
             
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="#"
              >
                Marketplace
              </a>
            </li>
             
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="#"
              >
                Explore
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
     
    <div
      class="col-md-1"
    >
      <img
        alt="user photo"
        src="https://avatars3.githubusercontent.com/u/39833045?s=30&v=4"
      />
    </div>
  </div>
</header>
`;
